actioncable/lib/action_cable/connection/subscriptions.rb

```diff
@@ -29,7 +29,9 @@ def add(data)
         subscription_klass = id_options[:channel].safe_constantize
 
         if subscription_klass && ActionCable::Channel::Base >= subscription_klass
-          subscriptions[id_key] ||= subscription_klass.new(connection, id_key, id_options)
+          subscription = subscription_klass.new(connection, id_key, id_options)
+          subscriptions[id_key] ||= subscription
+          subscription.registered!
```

Comment:
This allocates an extra instance of the `subscription_klass` needlessly if it's already there. How about:

```ruby
subscriptions[id_key] ||= subscription_klass.new(connection, id_key, id_options).
  tap(&:registered!)
```